src/main/java/teammates/ui/controller/AdminSessionsPageAction.java

```diff
@@ -117,25 +114,24 @@ private ActionResult createShowPageResultIfParametersInvalid() {
             SanitizationHelper.sanitizeForHtml(endMin);
             SanitizationHelper.sanitizeForHtml(timeZone);
 
-            zone = Double.parseDouble(timeZone);
+            zone = TimeHelper.convertToZoneId(Double.parseDouble(timeZone));
 
-            start = TimeHelper.convertLocalDateTimeToDate(
-                    TimeHelper.parseLocalDateTime(startDate, startHour, startMin));
-            end = TimeHelper.convertLocalDateTimeToDate(
-                    TimeHelper.parseLocalDateTime(endDate, endHour, endMin));
+            start = TimeHelper.parseLocalDateTime(startDate, startHour, startMin);
+            end = TimeHelper.parseLocalDateTime(endDate, endHour, endMin);
 
-            if (start.after(end)) {
+            if (start.isAfter(end)) {
```

Comment:
I think these nested if else are quite convoluted. Is it possible to simplify to improve readability?